test/runner/job_test.exs

```diff
@@ -45,4 +19,253 @@ defmodule ElixirBench.Runner.JobTest do
+                 "runner" => %{
+                   "image" => "elixirbench/runner:1.5.2-20.1.2",
+                   "command" => "mix run bench/bench_helper.exs",
+                   "network_mode" => "host",
+                   "environment" => %{
+                     "ELIXIRBENCH_REPO_SLUG" => "elixir-ecto/ecto",
+                     "ELIXIRBENCH_REPO_BRANCH" => "mm/benches",
+                     "ELIXIRBENCH_REPO_COMMIT" => "2a5a8efbc3afee3c6893f4cba33679e98142df3f"
+                   },
+                   "volumes" => [
+                     "/tmp/benchmarks/test_job:/var/bench:Z"
+                   ]
+                 }
+               },
+               "version" => "3"
+             } = compose_config
+    end
+
+    test "return docker-compose setup for job with deps and environment variables" do
+      job = job_fixture() |> with_deps |> with_env_variables
+      timeout = 200
+
+      compose_config =
+        Job.get_compose_config(job)
+        |> Antidote.decode!()
+
+      %{"services" => services} = compose_config
+
+      assert %{"job_test_job_mysql" => _mysql} = services
+      assert %{"job_test_job_postgres" => _postgres} = services
+
+      %{"runner" => %{"command" => command}} = services
+
+      assert command =~
+               "wait-for.sh localhost:3306 -t #{timeout} -- wait-for.sh localhost:5432 -t #{
+                 timeout
+               } -- mix run bench/bench_helper.exs"
+
+      assert %{"runner" => %{"depends_on" => ["job_test_job_mysql", "job_test_job_postgres"]}} =
+               services
+
+      assert %{"runner" => %{"environment" => %{"PG_URL" => "postgres:postgres@localhost"}}} =
+               services
+
+      assert %{"runner" => %{"environment" => %{"MYSQL_URL" => "root@localhost"}}} = services
+    end
+  end
+
+  describe "format_measurement/2" do
+    test "extract and format measurements information for benchmark given valid input" do
+      benchmark_name = "insert_mysql"
+      measurements = measurement_fixture()
+
+      %{
+        "statistics" => %{
+          "insert_changeset" => insert_changeset,
+          "insert_plain" => insert_plain
+        }
+      } = measurements
+
+      formatted_measurements = Job.format_measurement(measurements, benchmark_name)
+
+      assert %{"insert_mysql/insert_changeset" => ^insert_changeset} = formatted_measurements
+      assert %{"insert_mysql/insert_plain" => ^insert_plain} = formatted_measurements
+
+      measurements = Job.format_measurement(measurement_fixture(), "")
+
+      assert %{"/insert_changeset" => _} = measurements
+      assert %{"/insert_plain" => _} = measurements
+    end
+
+    test "return empty map given invalid inputs" do
+      assert %{} = Job.format_measurement(measurement_fixture(), nil)
+      assert %{} = Job.format_measurement(%{}, "insert_mysql")
+
+      assert %{} = Job.format_measurement(nil, "insert_mysql")
+      assert %{} = Job.format_measurement(%{"a" => "b"}, "insert_mysql")
+    end
+  end
```

Comment:
where are we testing that we drop all that big measurement data? It's also not quite sure for me where that happens in the code as it's not that explicit, so I think we could improve the code and tests there :)

Reply:
there is the `format_measurements/2` I added a comment on it to explicit say that we are interested in just the `statistics data`